spec/src/main/asciidoc/metrics.asciidoc

```diff
@@ -26,164 +26,188 @@ the methods annotated with a `@Retry`, `@Timeout`, `@CircuitBreaker`, `@Bulkhead
 === Names
 
 The automatically added metrics follow a consistent pattern which includes the fully qualified name of the annotated method.
-In the tables below, the placeholder `<name>` should be replaced by the fully qualified method name.
 
 If two methods have the same fully qualified name then the metrics for those methods will be combined. The result of this combination
 is non-portable and may vary between implementations. For portable behavior, monitored methods in the same class should have unique names.
 
+=== Scope
+
+Metrics added by this specification will appear in the `base` MicroProfile Metrics scope.
+
 === Metrics added for `@Retry`, `@Timeout`, `@CircuitBreaker`, `@Bulkhead` and `@Fallback`
 
 Implementations must ensure that if any of these annotations are present on a method, then the following metrics are added only once for that method.
 
-[cols="8,3,4,9"]
+[cols="1,5"]
 |===
-| Name | Type | Unit | Description
-
-|`ft.<name>.invocations.total`
-| Counter | None
-| The number of times the method was called
-
-|`ft.<name>.invocations.failed.total`
-| Counter | None
-| The number of times the method was called and, after all Fault Tolerance actions had been processed, threw a `Throwable`
+| Name | `ft.invocations.total`
+
+| Type | `Counter`
+| Unit | None
+| Description | The number of times the method was called
+| Tags
+a| * `method` - the fully qualified method name
+ * `result` = `[valueReturned\|exceptionThrown]` - whether the invocation returned a value or threw an exception
+ * `fallbackUsed` = `[true\|false\|none]` - `true` if fallback was used, `false` if a fallback is configured but was not used, `none` if a fallback is not configured
```

Comment:
Had a further thought on this:
fallbackUsed with the choice of `true|false|none` is odd. How about fallbackStatus - `true|false|none`

Reply:
I reviewed the whole section and I think it is better to change `none` to `notApplicable`

`fallbackUsed` =`[ true|false|notApplicable]`

Reply:
A fallback that is not applicable is a fallback that has been defined, but configured to not apply for particular situation. It's an interesting case that we don't cover, but it's distinct from fallback that is not defined (which is what `none` means now).

What we have now is, essentially: `fallback=[applied|notApplied|notDefined]`.

We could expand that to `fallback=[applied|notAppliedBecauseNotNeeded|notAppliedBecauseOfConfiguration|notDefined]`. Whether we need to distinguish the two cases of why fallback wasn't applied, I don't know.

Reply:
We already know whether it was not applied because it wasn't needed or was not applied because the exception didn't match the configuration by the value of `result`

We currently have
* `result = valueReturned, fallbackUsed = false` - method returned successfully, fallback was not needed
* `result = valueReturned, fallbackUsed = true` - method threw an exception, fallback was invoked and also threw an exception
* `result = exceptionThrown, fallbackUsed = false` - method threw an exception, but fallback was not invoked because fallback configuration didn't include the thrown exception
* `result = exceptionThrown, fallbackUsed = true` - method threw an exception, fallback was invoked and also threw an exception

Reply:
Ah, good point, thanks!

Reply:
What I wanted to say is that `notApplicable` can't be used instead of `none`. It could be `notDefined` or perhaps `undefined`, though `none` is fine in my opinion.

Reply:
+1 on `notDefined`

Reply:
I don't like `notDefined` - it sounds like there's something which could have a value but it hasn't been set, which isn't really the situation here.

`notApplicable` makes the most sense idiomatically - the whole question of whether fallback was used or not doesn't apply because there is no `@Fallback` annotation - but it has the downside of overlapping with the `applyOn` attribute and so could be misunderstood to mean "the fallback doesn't apply to this exception" rather than "the question of whether fallback was used or not does not apply".

`notEnabled` would be another option - this overlaps with the ability to disable and enable particular annotations via config, but maybe that's ok? A method where the annotation is disabled is functionally identical to one where the annotation is not present.

Reply:
`notEnabled` - does not mean the annotation does not exist. It could not the exception does not enable the fallback. Maybe `notSpecified` is better.

Reply:
I propose we stay with `none`. We use some really excessively long names elsewhere, and I'm not really a big fan of that.

Reply:
I like the long names you specified @Ladicek , as it is self explainary 👍 .

Reply:
> `notEnabled` - does not mean the annotation does not exist. It could not the exception does not enable the fallback.

I don't think we ever use the word "enabled" to talk about `applyOn` or `skipOn`, so I don't think it would be reasonable to think that it relates to that. We do use "enabled" to talk about enabling and disabling an annotation via config.

> EDIT: alternatively, I propose we go with long names here as well, and use something like fallback=[applied|notApplied|notDefined] :-)

I still don't really like `notSpecified` or `notDefined` as it sounds to me too much like a value which should exist is missing, but I can live with it. `fallback=notDefined` is at least better than `fallbackUsed=notDefined`.

Reply:
I can see how `undefined` could sound like "something is supposed to be here, but some programmer made a mistake somewhere and so you're seeing this placeholder" (especially for people who have seen JavaScript), but `notDefined` is pretty unique IMHO.

Reply:
I've changed it to `fallback=[applied|notApplied|notDefined]`